src/main/java/com/aws/greengrass/telemetry/TelemetryAgent.java

```diff
@@ -147,6 +147,10 @@ public TelemetryAgent(Topics topics, MqttClient mqttClient, DeviceConfiguration
         // Subscribe to thing name changes.
         deviceConfiguration.getThingName()
                 .subscribe((why, node) -> updateThingNameAndPublishTopic(Coerce.toString(node)));
+
+        if (!deviceConfiguration.isDeviceConfiguredToTalkToCloud()) {
+            this.isConnected.set(false);
```

Comment:
Same question as in FSS above.

Reply:
Telemetry data is emitted as log files and not cached in memory. We can probably also skip scheduled upload for the same reason as FSS.